src/OpenTelemetryEngine/Traces/ActivityExtensions.cs

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace OpenTelemetryEngine.Traces
+{
+    internal static class ActivityExtensions
+    {
+        private static readonly Action<Activity, string>? s_setSpanId;
+        private static readonly Action<Activity, string>? s_setId;
+        private static readonly Action<Activity, string>? s_setTraceId;
+        private static readonly Action<Activity, string>? s_setRootId;
+
+        static ActivityExtensions()
```

Comment:
I would love to understand what we are doing here and, more importantly, _why_ we are doing it. Is this to gain access to private `Activity` fields? This looks suspicious: are we doing something unconventional and/or unintended by the framework authors? Have we considered consequences and alternatives? If this actually turns out to be the best thing to do, it deserves an extensive explanation in the comments.

Reply:
Summarizing our discussion from yesterday in this format -
We are doing something that is outside of the "expected normal use case" for Activity here by not creating a new activity in the worker which is a child of the host activity, but actually duplicating the host span in its entirety. Because Activity does not have a constructor or a factory method which allows setting of the span id, id, trace id, and root id properties, we chose to override the behavior of the class in this way. 
There is some inherent risk here - if Activity's internal implementation changes (which it may at any time, because the owners of this class intended these fields to be private), it may break or change the behavior of our module in ways we can't predict. 
Will discuss further later today

Reply:
.NET team is aware of this approach and has encouraged us to continue. Discussion is continuing surrounding changing System.Diagnostics to allow for modifications of these properties, but this work will not be complete within the next few months. 
We should add a unit test for this case. https://github.com/Azure/azure-functions-powershell-opentelemetry/issues/14